packages/sit-onyx/src/components/OnyxCalendar/OnyxCalendar.vue

```diff
@@ -0,0 +1,358 @@
+<script lang="ts" setup>
+import { createCalendar } from "@sit-onyx/headless";
+import { iconChevronLeftSmall, iconChevronRightSmall } from "@sit-onyx/icons";
+import { computed, ref, useTemplateRef, watch } from "vue";
+import { useDensity } from "../../composables/density.js";
+import { useResizeObserver } from "../../composables/useResizeObserver.js";
+import {
+  SKELETON_INJECTED_SYMBOL,
+  useSkeletonContext,
+} from "../../composables/useSkeletonState.js";
+import { injectI18n } from "../../i18n/index.js";
+import { ONYX_BREAKPOINTS } from "../../utils/breakpoints.js";
+import OnyxHeadline from "../OnyxHeadline/OnyxHeadline.vue";
+import OnyxSkeleton from "../OnyxSkeleton/OnyxSkeleton.vue";
+import OnyxSystemButton from "../OnyxSystemButton/OnyxSystemButton.vue";
+import type { OnyxCalderProps } from "./types.js";
+
+const props = withDefaults(defineProps<OnyxCalderProps>(), {
+  size: "auto",
+  selection: "single",
+  weekStartDay: "Monday",
+  displayCalendarWeek: false,
+  disabled: false,
+  skeleton: SKELETON_INJECTED_SYMBOL,
+});
+/**
+ * Emit selectedDate on Change
+ */
+const emit = defineEmits(["update:selectedDate"]);
+
+defineSlots<{
+  /**
+   * Optional slot that is displayed below at the right of the Header.
+   */
+  actions?(): unknown;
+}>();
+const calendarSize = computed(() => {
+  return props.size !== "auto" ? props.size : width.value < ONYX_BREAKPOINTS.xs ? "small" : "big";
+});
+const { locale } = injectI18n();
+
+const dayNames = computed(() => {
+  const days = Array.from({ length: 7 }, (_, i) => {
+    const date = new Date(2024, 0, 1 + i);
+    return date;
+  });
+  const formatStyle = calendarSize.value === "big" ? "long" : "short";
+  const formatter = new Intl.DateTimeFormat(navigator.language, { weekday: formatStyle });
+  return days.map((day) => formatter.format(day));
```

Comment:
Please make use of the `injectI18n` composable:

```ts
const i18n = injectI18n();
```

```suggestion
  const formatter = new Intl.DateTimeFormat(i18n.locale { weekday: formatStyle });
  return days.map((day) => formatter.format(day));
```